### PR TITLE
Drag drop

### PR DESCRIPTION
**Pull request by sirherrbatka**
At this point it seems that everything works. Still, I think we should think about further improvements, maybe a panel holding recently dragged items someday?

### CODE REVIEW

**Comment by sirherrbatka** [comment]
Ah, one more thing: multiple documents are not handled right now.

File: apps/opencs/model/world/universalid.cpp [hunk 1]
```diff
@@ -186,7 +187,7 @@
             mClass = sIdArg[i].mClass;
             return;
         }
-
+    std::cout<<type<<std::endl;
```
Comment (zinnschlag) [inline apps/opencs/model/world/universalid.cpp]: Left over debugging instruction?

File: apps/opencs/model/world/tablemimedata.cpp [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+    throw ("TableMimeData object does not hold object of the seeked type");
+}
+
+CSMWorld::UniversalId::Type CSMWorld::TableMimeData::convertEnums (CSMWorld::ColumnBase::Display type) const
+{
+    switch (type)
+    {
+        case CSMWorld::ColumnBase::Display_Race:
+            return CSMWorld::UniversalId::Type_Race;
+            break;
```
Comment (zinnschlag) [inline apps/opencs/model/world/tablemimedata.cpp:124]: This break is completely pointless.

File: apps/opencs/model/world/tablemimedata.hpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
+/*This class provides way to construct mimedata object holding the reference to the
```
Comment (zinnschlag) [inline apps/opencs/model/world/tablemimedata.hpp]: Comments describing a class are better done with a doxygen comment immediately before the class.

File: apps/opencs/view/world/table.cpp [hunk 4]
```diff
@@ -13,9 +14,12 @@
 #include "../../model/world/idtable.hpp"
 #include "../../model/world/record.hpp"
 #include "../../model/world/columns.hpp"
+#include "../../model/world/tablemimedata.hpp"
 
 #include "recordstatusdelegate.hpp"
 #include "util.hpp"
+#include <qt4/QtCore/qnamespace.h>
```
Comment (zinnschlag) [inline apps/opencs/view/world/table.cpp]: Can we try to maintain include ordering? (see wiki)

File: apps/opencs/model/world/tablemimedata.cpp (hunk 2 above)
Comment (sirherrbatka) [inline apps/opencs/model/world/tablemimedata.cpp:124]: I just like break after case for the sake of clarity.

**Comment by zinnschlag** [comment]
Merge in progress. So what is missing?
- document-check
- drag to filter field
- drag of IDs that can be used in scripts into the script editor window

I think that is all for now, right? We can move the filter field to a separate issue, because that is going to be more complex and time consuming (probably not ready for OpenMW 0.29.0). The other two issues should be resolved, if possible.

**Comment by sirherrbatka** [comment]
I will start looking into remaing issues. Not sure how to handle document 
check but I suspect that I simply need to store path to the document in the 
TableMimeData. That being said I, need to have access to that path from the 
tableview. Well I will check this out.

**Comment by zinnschlag** [comment]
Can't you just use a pointer to the Data object or the Document? Both are unique and don't change during an editing session.

**Comment by sirherrbatka** [comment]
Actually I'm just looking at the document part of the codebase (didn't do this 
before, simply because I had to only deal with the records), and It looks like 
a correct idea. I will just encapsulate some pointer inside the tablemimedata.

**Pull request merged**